### ISSUE TITLE
tensorflow cosine similarity

### ISSUE THREAD

**Issue by beijinggao** [beijinggao/laogaoganhuo#3]
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/examples/tutorials/word2vec/word2vec_basic.py

**Comment by beijinggao** [comment]
optimizer = tf.train.GradientDescentOptimizer(1.0).minimize(loss)
--
# Compute the cosine similarity between minibatch examples and all embeddings.
norm = tf.sqrt(tf.reduce_sum(tf.square(embeddings), 1, keep_dims=True))
normalized_embeddings = embeddings / norm
valid_embeddings = tf.nn.embedding_lookup(
normalized_embeddings, valid_dataset)
similarity = tf.matmul(valid_embeddings, normalized_embeddings, transpose_b=True)
